### PR TITLE
Add runtime estimator

### PR DESCRIPTION
**Pull request by dstrain115**
- Adds a utility to estimate runtime for QCS
- Estimates single circuit runs, sweeps, and batches.
- This is a rough approximation and only accurate to about 25% or so.

### CODE REVIEW

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]
This is really impressive! In my experience fiddling with these numbers, consistently estimating within 50% of the target was a challenge, much less 25%.

The one thing I think is missing from this PR is documentation on how to reproduce these values for future devices. Happy to discuss this offline if it's not something we're ready to make public.

**Comment by dstrain115** [comment]
Added a line about how the rep rate calculator was used to get timings and that the model was fitted by hand.
Also added to the engine init file.

**Review by MichaelBroughton — COMMENTED** [review]
High level question:

Given your description of how tightly these estimates match reality (25%); how you drew the data by hand to fit the model (I'm guessing the numbers present in the test); and how all the functions seem to be based around a few basic variables `width`, `depth`, `n_qubits` etc.

What do we think about just turning this into a purely statistical set of functions ?
1. Fit a linear model on these variables.
2. Write out the functions to just be:
```python3
def run_sweep_time(circuit, ... ,...):
  _run_swee_internal(depth, weidth, etc.)

def _run_sweep_internal(x, y, z):
  return 0.3 * x + 2.9 * y - 4.8 * z
```
This seems like a process that might be a little more robust to internal changes (all we have to do is re-fit the coefficients and we've updated it to the latest behavior vs. having to rewrite more complicated code to bake in assumptions from the changed underlying process). What do you think ?

**Review by 95-martin-orion — COMMENTED** [review]
My comments have been addressed - leaving to @MichaelBroughton for final approval.

File: cirq-google/cirq_google/engine/runtime_estimator_test.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        (40, 80, 32000, 16, 90.4),
+        (40, 80, 32000, 8, 58.7),
+        (40, 80, 8000, 32, 80.1),
+        (20, 40, 32000, 32, 69.8),
+        (30, 40, 32000, 32, 170.9),
+        (40, 40, 32000, 32, 215.4),
+        (2, 40, 16000, 16, 10.5),
+        (2, 640, 16000, 16, 16.9),
+        (2, 1280, 16000, 16, 22.6),
+        (2, 2560, 16000, 16, 38.9),
```
[the remark this answers is not in the document]
Comment (dstrain115) [reply cirq-google/cirq_google/engine/runtime_estimator_test.py:103]: Yes.

**Comment by dstrain115** [comment]
> This seems like a process that might be a little more robust to internal changes (all we have to do is re-fit the coefficients and we've updated it to the latest behavior vs. having to rewrite more complicated code to bake in assumptions from the changed underlying process). What do you think ?

We could try to fit a linear model.  I think there are some step functions in here that might not be captured well that I managed to work around in my hand-fitting.  I think the function I wrote though is mostly linear though, so I can give it a try.

**Review by MichaelBroughton — APPROVED** [review]
LGTM

**Comment by dstrain115** [comment]
Update: linear model did not fit very well.  It is at least twice as bad as the hand-tuned model that I used.  It also has a intercept which marks many data points as negative amounts of time, which is not ideal.  Though it will be much harder to update this hand-tuned model, I recommend sticking with it for now until an (as yet unforeseen) need for frequent updating causes us to reevaluate this approach.

**Pull request merged**